enhancements/split_helm_ansible_into_their_repos.md

```diff
@@ -0,0 +1,187 @@
+|  operator-sdk/test/ansible-memcached/	|  ansible-operator/e2e-tests/ansible-memcached/ |   has the static mock data which still used by molecule tests	|
+|  operator-sdk/test/ansible/	|  ansible-operator/e2e-tests/ansible/ |   has the static mock data which still used by molecule tests	|
+
+**NOTE**: The molecule tests still via shell script, however, if they be migrated to go test stack then the mock static files will no longer be required. 
+
+### SDK Operator lib
+
+The idea of this project as described above is to centralize all code which is useful for all types projects. In this way, SDK CLI as any operator type repo can import this lib and use its features. 
+ 
+In this way follow up the SDK tree with the points that would be moved:
+
+- `internal/util/k8sutil/constants.go`: Just the `WatchNamespaceEnvVar` should be moved since it is used for both types in the `run.go` implementation and not used by the SDK CLI commands at all. 
+- `internal/plugins`: code implementation which is used to customize the scaffold of the projects and allow they work on and be integrated with the framework features (plugins phase 2). Note that ansible/helm/go plugins will not be moved to this repository.    
+- `test/internal`: the utils tests implemented in SDK can will used for any operator type and SDK CLI repo as well to implement the e2e tests.  
+
+### Usage of Helm/Ansible plugin in SDK repo 
+
+Their usage in the SDK repo will be done as it currently is made for the Go operator. See: https://github.com/operator-framework/operator-sdk/tree/master/internal/plugins/golang/v2
+
+## Test Plan
+
+The ultimate test is that e2e tests and manual testing of the SDK binary will
+be used to determine it remains functional. Each repo will contain their own e2e and unit tests which ensure that they all will still working as should be after the split.
+ 
+Besides Ansible/Helm be decoupled from SDK CLI tool their tests should still ensure their functionalities with the tool which means that we will need to test them with the SDK binary provide from master. 
+
+Also, to avoid the risk and scenarios where the changes in the tool can impact in their behaviours and functionalities still required have in SDK e2e test which will scaffold the projects for each type and test the tool integration  with them such as to ensure scorecard and OLM features. 
+
+In this way we will have:
+
+- Ansible/Helm repositories with unit and e2e tests 
+- SDK repository with the pieces of the e2e tests which ensure its features
+- SDK Operator lib covered with unit tests and tested indirectly as well via the SDK/ Ansible/Helm repo since the lib will be imported and used by them.  
+
+## Examples
+
+The project [joelanford/helm-operator](https://github.com/joelanford/helm-operator) is an excellent example to illustrate and demonstrate its motivations and the expected final result. Note that it is not using the SDK Operator lib as other granular details described here.   
```

Comment:
We say that using this operator is a non-goal, but then reference it in examples? I'd rather not talk about it at all in that case.